### PR TITLE
Get user’s lat/lng and display location on the Map

### LINKED ISSUE #5: #3: Get user’s lat/lng
Modern browsers provide a `navigator.geolocation.watchPosition` API that takes care of prompting the user to use their location then continuously feeds the user’s (well, the _device’s_ tbh) location to a callback function.

AC:

1. User should hit a button in the UI somewhere that initiates the call to `navigator.geolocation.watchPosition`
2. Once the user clicks the button, the button should change to disabled and the button text should change to something that indicates that the user has already shared their location
3. Assuming the user clicks “Allow” on their device, a pulsing blue dot should appear on the map at the user’s location
4. The map should pan to center the blue dot

### PR DESCRIPTION
**Pull request by AshlingH**
## Description

User is able to click on "Share Location?" button. Once clicked, the button is changed to a disabled "Location Shared" state. User is prompted to allow the app to access their location by the browser window. Once user clicks allow, the latitude and longitude are shared, and both the pulsating blue dot and the map adjust to the user's location.

## Related Issue

Closes out #5. 

## Acceptance Criteria

- [x] User should hit a button in the UI somewhere that initiates the call to navigator.geolocation.watchPosition

- [x] Once the user clicks the button, the button should change to disabled and the button text should change to something that indicates that the user has already shared their location

- [x] Assuming the user clicks “Allow” on their device, a pulsing blue dot should appear on the map at the user’s location

- [x] The map should pan to center the blue dot

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

#### EXAMPLE 1 - DEFAULT LOAD / PRIOR TO SHARING LOCATION
Screen Shot 2020-05-19 at 6 32 16 PM
#### EXAMPLE 2 - LOCATION SHARED + BLUE DOT AND MAP UPDATED
Screen Shot 2020-05-19 at 6 31 59 PM

### CODE REVIEW

File: src/components/Footer.js [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+      navigator.geolocation.getCurrentPosition(success, error, options);
+   }
+
 
 const Footer= () => {
     return (
         <footer className= "footer">
             
-        <button>Share Location</button>
+        <button onClick={() => handleClick ()}>Share Location?</button>
```
Comment (segdeha) [inline src/components/Footer.js]: Your `onClick` handler can be simplified to:

    onClick={ handleClick }

**Comment by mjhcodes** [comment]
For fun, I adjusted the coordinates of the default location to Cork, Ireland, in honor of one of our team members. This can 100% be changed again, but thought it was better than Russia? 🤷‍♂️

**Comment by AshlingH** [comment]
Lovely to see Cork!!Great use of props, looking forward to hearing about how you all brought it together. The geolocation call(api) didn't work for me but did work in incognito mode!

File: src/App.css [hunk 2]
```diff
@@ -21,3 +21,36 @@
   display:block;
   text-align: center;
 }
+
+
+.MyLocation {
+  animation: pulseDot 1.25s cubic-bezier(0.455, 0.03, 0.515, 0.955) -.4s infinite;
```
Comment (jendevelops) [inline src/App.css]: So cool!

File: src/App.js [hunk 3]
```diff
@@ -1,16 +1,54 @@
...
+import React, { useState } from "react";
 import "./App.css";
 import Header from "./components/Header";
-import Footer from "./components/Footer";
 import Main from "./components/Main";
+import Footer from "./components/Footer";
+
 function App() {
+  const [marker, allowMarker] = useState(false);
+  const [data, setData] = useState({
```
Comment (jendevelops) [inline src/App.js]: ~~I wonder if we can rename this state to be a little more descriptive. We will have quite a few state names in this app, but a name like mapCenter and setMapCenter might give more context.~~

I see that data is more than just the lat, lng of the center from line 32. Could we put a comment in explaining what it is? That way anyone looking at the code will always have context to what lives inside data state.

File: src/components/Main.js [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
+    <section className="main">
+      <Map data={data} />
+    </section>
+  );
+};
 
-export default Main;
\ No newline at end of file
+Main.propTypes = {
+  data: PropTypes.object.isRequired,
```
Comment (jendevelops) [inline src/components/Main.js]: Nice type checking. Could you explain the advantage of using the prop-types package as opposed to just checking the type and throwing an error within the code? (We can talk about it on Sunday as well!)

File: src/components/Map.js [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
 
-function Map(){
-    const defaultLocation = {
-        center:{
-            lat: 59.95,
-            lng: 30.33
-        },
-        zoom: 11
+function Map(props) {
+  const { data } = props;
```
Comment (jendevelops) [inline src/components/Map.js]:
```suggestion
function Map({data}) {
```
This simplifies the code by unpacking the props and assigning to a variable within the parameter.

File: src/components/Main.js [hunk 6]
```diff
@@ -1,13 +1,19 @@
 import React from "react";
-import Map from '../components/Map';
+import PropTypes from "prop-types";
+import Map from "../components/Map";
 
+const Main = (props) => {
+  const { data } = props;
```
Comment (jendevelops) [inline src/components/Main.js]:
```suggestion
const Main = ({data}) => {
```
This simplifies the code by unpacking the props and assigning to a variable within the parameter.

**Review by jendevelops — APPROVED** [review]
Looking at the code I have one larger suggestion to make is to change the passing of data state down two levels into Map. In our most recent PR, we extracted the definition of the map component at the App level, and pass down the map component itself into main. 
```
function App() {
  const [nearbyPlaces, setNearbyPlaces] = useState([]);
  const [currentPin,setCurrentPin] = useState({title:null, description:null, image:null});
  const [displayInformation, setDisplayInformation] = useState(false);

  const map = <Map nearbyPlaces={nearbyPlaces} setNearbyPlaces={setNearbyPlaces} setCurrentPin={setCurrentPin} setDisplayInformation={setDisplayInformation} />


  return (
    <div className="App">
      ...
      <Main map={map} />
      ...
    </div>
  );
}
```
The component as a prop is then rendered in Main like this:
```
const Main = ({map}) => (
  <section className="main">
    {map}
  </section>
);
```

Since we are already doing that, this might benefit this PR as well to pass data into the defined Map component, and pass the component itself as a prop. This technique was suggested in the bottom of the Context documentation in a section called [_Before You Use Context_](https://reactjs.org/docs/context.html#before-you-use-context). Definitely something we can talk about on Sunday!

File: src/App.css (hunk 2 above)
Comment (segdeha) [reply src/App.css]: Agreed, this is a super cool effect!

File: src/App.js (hunk 3 above)
Comment (segdeha) [reply src/App.js]: `data` is a really generic variable name. Jen's suggestion of a comment is a good one, but even better would be a name that's more descriptive. That way, when you see the variable in other places in the app, we won't have to refer back to this to understand what it represents.

File: src/components/Footer.js [hunk 7]
```diff
@@ -1,22 +1,48 @@
 import React from "react";
+import PropTypes from "prop-types";
 
+const Footer = (props) => {
+  const { handleClick, marker } = props;
 
+  return (
+    <footer className="footer">
+      <h1>
+        {" "}
```
Comment (segdeha) [inline src/components/Footer.js]: What does this space do here?

File: src/components/Footer.js [hunk 8]
```diff
@@ -1,22 +1,48 @@
...
-        <div>
-        <p><strong> Click "Share Location"</strong> to discover interesting landmarks around you from Wikipedia</p>
-        <p><strong>Tap on a marker</strong> on the map and more information about it will show up in this space. Have fun!</p>
-        <p>Note: Your location is never stored in the app or on our servers!</p>
-        <p><strong> What's near me?</strong> Looks and works best when installed on your phone as an app.</p>
-        </div>
-       
+      <div>
+        <p>
+          {" "}
```
Comment (segdeha) [inline src/components/Footer.js]: Help me understand why these spaces are necessary....

File: src/components/Footer.js [hunk 9]
```diff
@@ -1,22 +1,48 @@
...
+        </p>{" "}
+      </div>
+    </footer>
+  );
+};
 
-        </footer>
-    )
-}
+Footer.propTypes = {
```
Comment (segdeha) [inline src/components/Footer.js:34]: Nice! Love that we're using `propTypes`!

File: src/components/Main.js (hunk 6 above)
Comment (segdeha) [reply src/components/Main.js]: Nice one, Jen!

File: src/components/Map.js (hunk 5 above)
Comment (segdeha) [reply src/components/Map.js]: 👍

File: src/components/MyLocation.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import "../App.css";
+
+//const MyLocation = () => <span className="MyLocation" role="img" aria-label="My location">⚫</span>;
```
Comment (segdeha) [inline src/components/MyLocation.js]: Remove this comment if it's not needed

File: src/index.js [hunk 11]
```diff
@@ -5,9 +5,12 @@ import App from './App';
 import * as serviceWorker from './serviceWorker';
 
 ReactDOM.render(
-  <React.StrictMode>
+  // <React.StrictMode>
```
Comment (segdeha) [inline src/index.js]: Please remove these comments before merging. Thanks!

**Review by segdeha — APPROVED** [review]
This looks and works great! There are a handful of suggestions in the comments, but nothing I can see that's blocking. Good work! 👍

**Comment by segdeha** [comment]
Looks like you need to merge `master` into the branch and resolve conflicts in order to merge

File: src/App.js (hunk 3 above)
Comment (mjhcodes) [reply src/App.js]: Good call. I changed this to `mapProperties`. Hope that helps with the clarification of what the variable is and contains at other component levels.

File: src/components/Main.js (hunk 4 above)
Comment (mjhcodes) [reply src/components/Main.js]: I'm not an expert on type checking, but I think the latter approach would be a little more cumbersome, as prop-types essentially does that for you and in a very concise, React friendly kind of way? Maybe Andrew can share a bit more insight / perspective on it.

File: src/components/Map.js (hunk 5 above)
Comment (mjhcodes) [reply src/components/Map.js]: Smooth. I like it. :-)

File: src/components/Footer.js (hunk 7 above)
Comment (mjhcodes) [reply src/components/Footer.js]: I'm thinking this was a "prettier" addition to force a space for visual purposes. I've removed them though and don't see any ill effects, so all good.

**Pull request merged**